clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -896,19 +896,31 @@ StringRef SYCL::gen::resolveGenDevice(StringRef DeviceName) {
           .Cases("intel_gpu_whl", "intel_gpu_9_5_0", "whl")
           .Cases("intel_gpu_aml", "intel_gpu_9_6_0", "aml")
           .Cases("intel_gpu_cml", "intel_gpu_9_7_0", "cml")
-          .Cases("intel_gpu_icllp", "intel_gpu_11_0_0", "icllp")
-          .Cases("intel_gpu_ehl", "intel_gpu_jsl", "ehl")
-          .Cases("intel_gpu_tgllp", "intel_gpu_12_0_0", "tgllp")
-          .Case("intel_gpu_rkl", "rkl")
-          .Cases("intel_gpu_adl_s", "intel_gpu_rpl_s", "adl_s")
-          .Case("intel_gpu_adl_p", "adl_p")
-          .Case("intel_gpu_adl_n", "adl_n")
+          .Cases("intel_gpu_icllp", "intel_gpu_icl", "intel_gpu_11_0_0",
+                 "icllp")
+          .Cases("intel_gpu_ehl", "intel_gpu_jsl", "intel_gpu_11_2_0", "ehl")
+          .Cases("intel_gpu_tgllp", "intel_gpu_tgl", "intel_gpu_12_0_0",
+                 "tgllp")
+          .Cases("intel_gpu_rkl", "intel_gpu_12_1_0", "rkl")
+          .Cases("intel_gpu_adl_s", "intel_gpu_rpl_s", "intel_gpu_12_2_0",
+                 "adl_s")
+          .Cases("intel_gpu_adl_p", "intel_gpu_12_3_0", "adl_p")
+          .Cases("intel_gpu_adl_n", "intel_gpu_12_4_0", "adl_n")
           .Cases("intel_gpu_dg1", "intel_gpu_12_10_0", "dg1")
-          .Cases("intel_gpu_acm_g10", "intel_gpu_dg2_g10", "acm_g10")
-          .Cases("intel_gpu_acm_g11", "intel_gpu_dg2_g11", "acm_g11")
-          .Cases("intel_gpu_acm_g12", "intel_gpu_dg2_g12", "acm_g12")
-          .Case("intel_gpu_pvc", "pvc")
-          .Case("intel_gpu_pvc_vg", "pvc_vg")
+          .Cases("intel_gpu_acm_g10", "intel_gpu_dg2_g10", "intel_gpu_12_55_8",
+                 "acm_g10")
+          .Cases("intel_gpu_acm_g11", "intel_gpu_dg2_g11", "intel_gpu_12_56_5",
+                 "acm_g11")
+          .Cases("intel_gpu_acm_g12", "intel_gpu_dg2_g12", "intel_gpu_12_57_0",
+                 "acm_g12")
+          .Cases("intel_gpu_pvc", "intel_gpu_12_60_7", "pvc")
+          .Cases("intel_gpu_pvc_vg", "intel_gpu_12_61_7", "pvc_vg")
+          .Cases("intel_gpu_mtl_u", "intel_gpu_mtl_s", "intel_gpu_arl_u",
+                 "intel_gpu_arl_s", "intel_gpu_12_70_4", "mtl_u")
+          .Cases("intel_gpu_mtl_h", "intel_gpu_12_71_4", "mtl_h")
+          .Cases("intel_gpu_arl_h", "intel_gpu_12_74_4", "arl_h")
+          .Cases("intel_gpu_bmg_g21", "intel_gpu_20_1_4", "bmg_g21")
+          .Cases("intel_gpu_lnl_m", "intel_gpu_20_4_4", "lnl_m")
```

Comment:
Could you add some test entries for the new values in `clang/test/Driver/sycl-oneapi-gpu-intelgpu.cpp` ?

Reply:
Thanks, added in https://github.com/intel/llvm/pull/13520/commits/b9492ceaf7f17bbd951c2f0651e71c4a1226830e